### PR TITLE
docs: add more optional search parameters to open api

### PR DESCRIPTION
**Pull request by Roxie-32**
Add more query parameters to the search open api

### CODE REVIEW

**Review by alexgarel — CHANGES_REQUESTED** [review]
Thanks for the work. Still I propose we improve a bit to make it more useful

File: docs/reference/api.yml [hunk 1]
```diff
@@ -351,7 +365,106 @@ components:
...
+      description: |
+        The origins_tags of product(s) you are searching for.
+    packaging_tags_de:
+      schema:
+        type: string
+        example: 1-jar-aus-klarglas
+      in: query
+      name: packaging_tags_de
+      description: |
+          The packaging_tag in german of product(s) you are searching for.
```
[the remark this answers is not in the document]
Comment (Roxie-32) [reply docs/reference/api.yml:451]: @alexgarel you were meant to explain this bette please

File: docs/reference/api.yml [hunk 2]
```diff
@@ -351,7 +365,106 @@ components:
...
+      name: states_tags_en
+      description: |
+        The states_tags in english of product(s) you are searching for.
+    traces_tags:
+      schema:
+        type: string
+      in: query
+      name: traces_tags
+      description: |
+        The traces_tags of product(s) you are searching for.
```
[the remark this answers is not in the document]
Comment (Roxie-32) [reply docs/reference/api.yml]: Fixed !

File: docs/reference/api.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (stephanegigandet) [reply docs/reference/api.yml:458]: @alexgarel very cool indeed :)

[the remark this answers is not in the document]
Comment (stephanegigandet) [reply docs/reference/api.yml:458]: > "will search in the `tag_name` tags in the language corresponding to `language_code"

It's a bit different: the language code applies to the values that are passed to the field. e.g. search for categories_tags_fr=moutarde is turned into a search for categories_tags=en:mustard

[the remark this answers is not in the document]
Comment (alexgarel) [reply docs/reference/api.yml:458]: @funke, taking into account @stephanegigandet comment:

```yaml
    tagname_with_language_code:
      description: |
        You can add a language code to a specific tag to query it in a specific language
      schema:
        type: Object
        patternProperty:
          (?<tag_name>\w+)_tags_(?<language_code>\w\w): 
            type: string
            description: will search in the `tag_name` tags using `language_code` translation of the tag. tag_name can be: origins, labels, countries, categories, etc. (refer to available parameters ending with `_tags`)
      examples:
        - packaging_tags_de: Flasche      
        - origins_tags_fr: Suisse
 ```

- It adds a parameter, which name is only descriptive (as the real name is defined by the patternProperty inside it), think of this name as an identifier for this type of parameter, so I used `tagname_with_language_code`
- then we use a schema definition of type object (it does not mean the parameter contains an object, it means we define the parameter using an object schema, it's a bit subtle, one as to read [the specification and it's examples](https://spec.openapis.org/oas/latest.html#fixed-fields-9) to really understand that). In our case, the property of the object, would indeed be the name the parameter takes.
- patternProperty enables us to use a regular expression to define property name (which, as stated above, becomes our parameter name). The expression, says we have a first part composed of multiple letters (\w) which is the tag_name (I used capturing expressions, [not sure it's really supported](https://json-schema.org/understanding-json-schema/reference/regular_expressions.html)), then _tags_ then the language code: two letters. If it does not work try with `(\w+)_tags_(\w\w)`

File: docs/reference/api.yml [hunk 3]
```diff
@@ -351,7 +365,106 @@ components:
...
-        The category of products you are searching for.
+        The category of product(s) you are searching for.
+    states_tags_en:
+      schema:
+        type: string
+        example: nutrition-facts-completed
+      in: query
+      name: states_tags_en
+      description: |
+        The states_tags in english of product(s) you are searching for.
```
[the remark this answers is not in the document]
Comment (stephanegigandet) [reply docs/reference/api.yml:377]: The states_tags exist on the public platform as well: http://world.openfoodfacts.org/states

File: docs/reference/api.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Roxie-32) [reply docs/reference/api.yml:458]: @alexgarel, this throws some errors, like the example should match the data type declared. I haven't understood the concept entirely, so its a bit difficult for me. Can you help out?

[the remark this answers is not in the document]
Comment (Roxie-32) [reply docs/reference/api.yml:458]: HI @alexgarel briniging your attention back to this.

**Review by alexgarel — APPROVED** [review]
I did a commit.

Seems like redoc as some hard time rendering the patternProperty, but I think we can merge as is for the moment.

See https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/openfoodfacts/openfoodfacts-server/search-parameters-api-docs/docs/reference/api.yml#tag/Read-Requests/operation/get-search

**Pull request merged**